src/pages/signin/LoginForm.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import {Text, TextInput, View} from 'react-native';
+import styles from '../../styles/styles';
+import SubmitButton from './SubmitButton';
+import openURLInNewTab from '../../libs/openURLInNewTab';
+import {hasAccount} from '../../libs/actions/Session';
+
+class LoginForm extends React.Component {
+    constructor() {
+        super();
+
+        this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
+
+        this.state = {
+            formError: false,
+            login: '',
+            isLoading: false,
+        };
+    }
+
+    /**
+     * Check that all the form fields are valid, then trigger the submit callback
+     */
+    validateAndSubmitForm() {
+        if (!this.state.login.trim()) {
+            this.setState({formError: 'Please fill out all fields'});
+            return;
+        }
+
+        this.setState({
+            formError: null,
+            isLoading: true,
+        });
+
+        // Check if this login has an account associated with it or not
+        hasAccount(this.state.login);
+    }
+
+    render() {
+        return (
+            <>
+                <View style={[styles.mb4]}>
```

Comment:
NAB, this form component could maybe be DRY-d up into a component like `SignInPanel` or something